ckanext/data_qld/reporting/controller.py

```diff
@@ -0,0 +1,278 @@
+import logging
+import ckan.plugins.toolkit as toolkit
+
+from ckan.lib.base import BaseController
+from ckanext.data_qld.reporting import constants
+from ckanext.data_qld.reporting.helpers import export_helpers, helpers
+
+log = logging.getLogger(__name__)
+get_action = toolkit.get_action
+request = toolkit.request
+render = toolkit.render
+get_validator = toolkit.get_validator
+ObjectNotFound = toolkit.ObjectNotFound
+Invalid = toolkit.Invalid
+NotAuthorized = toolkit.NotAuthorized
+abort = toolkit.abort
+_ = toolkit._
+DATAREQUEST_OPEN_MAX_DAYS = constants.DATAREQUEST_OPEN_MAX_DAYS
+COMMENT_NO_REPLY_MAX_DAYS = constants.COMMENT_NO_REPLY_MAX_DAYS
+
+
+class ReportingController(BaseController):
+
+    @classmethod
+    def check_user_access(cls):
+        toolkit.check_access(
+            'has_user_permission_for_some_org',
+            helpers.get_context(), {'permission': 'create_dataset'}
+        )
+
+    def index(self):
+        try:
+            self.check_user_access()
+
+            start_date, end_date = helpers.get_report_date_range(request)
+            org_id = request.GET.get('organisation', None)
+
+            organisations = helpers.get_organisation_list()
+
+            if organisations and len(organisations) == 1:
+                org_id = organisations[0]['value']
+
+            extra_vars = {
+                'organisations': organisations,
+                'user_dict': get_action('user_show')({}, {'id': toolkit.c.userobj.id})
```

Comment:
This is likely to fail because it provides an empty context dictionary and we don't allow anonymous access to `user_show`.

Reply:
You can send an empty context dictionary to an action via the toolkit.get_action method and it will automatically utilise the current context.

This is documented in the CKAN documentation:

https://docs.ckan.org/en/2.8/extensions/plugins-toolkit.html#ckan.plugins.toolkit.ckan.plugins.toolkit.get_action

"As the context parameter passed to an action function is commonly:"

```
context = {'model': ckan.model, 'session': ckan.model.Session,
           'user': pylons.c.user}
```

"an action function returned by `get_action()` will automatically add these parameters to the context if they are not defined. This is especially useful for plugins as they should not really be importing parts of ckan eg `ckan.model` and as such do not have access to `model` or `model.Session`."